commit 325496dfe42a
Author: Nicolas B. Pierron <nicolas.b.pierron@gmail.com>
Date:   Thu May 31 14:22:14 2018 +0200

    Bug 1464829 - Ensure the recover instruction vector has the expected size. r=jandem, a=jcristau
    
    --HG--
    extra : source : 931f3ee17589096df1e87223362051c0aec26931
    extra : intermediate-source : eab56078999a914abc3cf9ac16ca2ca7c06548d8
---
 js/src/jit/JitFrameIterator.h | 2 --
 js/src/jit/JitFrames.cpp      | 4 +---
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git js/src/jit/JitFrameIterator.h js/src/jit/JitFrameIterator.h
index ba5efef6a5438..3620badbda07e 100644
--- js/src/jit/JitFrameIterator.h
+++ js/src/jit/JitFrameIterator.h
@@ -322,9 +322,7 @@ class RInstructionResults
 
     MOZ_MUST_USE bool init(JSContext* cx, uint32_t numResults);
     bool isInitialized() const;
-#ifdef DEBUG
     size_t length() const;
-#endif
 
     JitFrameLayout* frame() const;
 
diff --git js/src/jit/JitFrames.cpp js/src/jit/JitFrames.cpp
index 646442b4c0358..dbb080e3d08d7 100644
--- js/src/jit/JitFrames.cpp
+++ js/src/jit/JitFrames.cpp
@@ -1621,13 +1621,11 @@ RInstructionResults::isInitialized() const
     return initialized_;
 }
 
-#ifdef DEBUG
 size_t
 RInstructionResults::length() const
 {
     return results_->length();
 }
-#endif
 
 JitFrameLayout*
 RInstructionResults::frame() const
@@ -2083,7 +2081,7 @@ SnapshotIterator::initInstructionResults(MaybeReadFallback& fallback)
     }
 
     MOZ_ASSERT(results->isInitialized());
-    MOZ_ASSERT(results->length() == recover_.numInstructions() - 1);
+    MOZ_RELEASE_ASSERT(results->length() == recover_.numInstructions() - 1);
     instructionResults_ = results;
     return true;
 }
